### PR TITLE
Davis6jm jovial devloper code review2

### PR DESCRIPTION
**Pull request by Jovial-Developer**
Description of the project:
This is an app that is a simple weather retrieval app using GPS navigation

What I learned from this code review.

    I learned how to help with design choices affecting accessibility for the user
    I learned to use an interface enforcing Dependency Inversion principle in SOLID
    I learned how to work with the Preview feature for Jetpack Compose
    I learned how to work with a GPS Application for the first time

This Project was available on GitHub on time, This program did build. This program did compile. This program does run on an emulator/physical phone Program does not have many comments, but variables are descriptive enough to understand the code. This project follows Kotlin coding conventions. More will descriptions will be added in pull request.

### CODE REVIEW

File: app/src/main/java/com/example/checkpoint/LocationPermissionsHelper.kt [hunk 1]
```diff
@@ -24,6 +24,10 @@ class LocationPermissionHelper(val activity: WeakReference<Activity>) {
                 override fun onPermissionResult(granted: Boolean) {
                     if (granted) {
                         onMapReady()
+                        Toast.makeText(
+                            activity.get(), "Using your current location. ",
+                            Toast.LENGTH_SHORT
```
Comment (Jovial-Developer) [inline app/src/main/java/com/example/checkpoint/LocationPermissionsHelper.kt:29]: Let the user know that their location is being used by the app for Accessibility

File: app/src/main/java/com/example/checkpoint/MainActivity.kt [hunk 2]
```diff
@@ -223,7 +225,9 @@ fun CheckpointHome(mapView: MapView){
         scope.launch {
             if (scaffoldState.bottomSheetState.isCollapsed) {
                 scaffoldState.bottomSheetState.expand()
-            } else {
+            }
+
```
Comment (Jovial-Developer) [inline app/src/main/java/com/example/checkpoint/MainActivity.kt:229]: Attempted to write code for collapsing bottom sheet scaffold by clicking on the text. It works well for expanding currently, but I think clicking on the label will help it close.  This is another accessibility thing.

File: app/src/main/java/com/example/checkpoint/MainActivity.kt [hunk 3]
```diff
@@ -268,7 +272,7 @@ fun BottomSheetContentSmall() {
 fun BottomSheetContentLarge() {
     Text(text = "Bottom Sheet Content Large",
         modifier = Modifier.padding(16.dp),
-        style = MaterialTheme.typography.h6,
+        style = MaterialTheme.typography.h4,
```
Comment (Jovial-Developer) [inline app/src/main/java/com/example/checkpoint/MainActivity.kt:275]: Expanding the text as the user draws up their view might be a good idea. so when the bottom sheet scaffold is expanded the text is slightly larger.

File: app/src/main/java/com/example/checkpoint/services/WeatherService.kt [hunk 4]
```diff
@@ -8,8 +8,12 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
-class WeatherService {
-    suspend fun fetchWeather(): WeatherAPI{
+interface IWeatherService {
+    suspend fun fetchWeather(): WeatherAPI
+}
+
```
Comment (Jovial-Developer) [inline app/src/main/java/com/example/checkpoint/services/WeatherService.kt:14]: SOLID Principle D Dependency Inversion states that you should implement and interface before a direct class.
https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#dependency-inversion-principle

**Comment by Jovial-Developer** [comment]
https://github.com/Jovial-Developer/IT3048CSpringProject/commit/ec452aef8346a182da349f226321562b717daf0f

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/efbc3818ac868c546c2bc7cea94bf4982756392c

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/efbc3818ac868c546c2bc7cea94bf4982756392c

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/82945b1080ef625b9ce20bfd2650d05cf22702c0

File: app/src/main/java/com/example/checkpoint/LocationPermissionsHelper.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/example/checkpoint/LocationPermissionsHelper.kt:29]: Avoid over-toasting.  That annoys the user.  
The user will know it's his/her current location by looking at the map.

File: app/src/main/java/com/example/checkpoint/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/checkpoint/MainActivity.kt:229]: I don't see any difference here, beyond the curlies being on another line.  ???

File: app/src/main/java/com/example/checkpoint/dto/Weather.kt [hunk 5]
```diff
@@ -5,4 +5,9 @@ data class Weather (
     val main: String,
     val description: String,
     val icon: String
-)
\ No newline at end of file
+){
+    private var CurrentWeather = main + " " + id + " " + description + " " + icon
```
Comment (discospiff) [inline app/src/main/java/com/example/checkpoint/dto/Weather.kt:9]: Use Kotlin shortcut "$main $id $description $icon"

**Comment by discospiff** [comment]
From this review:
- Using an interface is a good idea.
- Avoid over-toasting the user.
- Formatting can be handled automatically, and consistently, by the IDE.

**Pull request closed without merging**